datafusion/expr/src/type_coercion/aggregates.rs

```diff
@@ -480,7 +480,7 @@ pub fn is_avg_support_arg_type(arg_type: &DataType) -> bool {
         _ => matches!(
             arg_type,
             arg_type if NUMERICS.contains(arg_type)
-                || matches!(arg_type, DataType::Decimal128(_, _))
+                || matches!(arg_type, DataType::Decimal128(_, _)| DataType::Decimal256(_, _))
```

Comment:
Doesn't this change imply that people can run `AVG(dec256_col)` now? I tried to run this query


```sql
create table t as values (arrow_cast(123, 'Decimal256'));
select AVG(column1) from t;
```

But it seems like datafusion support for Decimal256 is still in progress

```
❯ create table t as values (arrow_cast(123, 'Decimal256(10,3)'));
Optimizer rule 'simplify_expressions' failed
caused by
This feature is not implemented: Can't create a scalar from array of type "Decimal256(10, 3)"
```

Reply:
hmm yes, I was under the impression that it should. In Dask-Sql's case we are not using the physical planner from DataFusion and rather using our own so maybe this error is coming in from some similar missed `Decimal256` coverage there? That would also explain why I don't see this issue when running as a library with dask-sql I guess?

Reply:
`simplify_expressions` is an optimization pass on the logical plan.
I suppose you're not using those or having your own.

Anyway, would be good to have some more support / (end-to-end) tests for `Decimal256`.